modules/fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/fs2/io/Metrics.scala

```diff
@@ -119,11 +126,25 @@ object Metrics {
 
       def badCount: F[Unit] =
         Sync[F].delay(badCounter.inc())
+    }
 
-      private def getGauge(now: Long, collectorTstamp: Long): Gauge[Long] =
-        new Gauge[Long] {
-          def getValue: Long = now - collectorTstamp
+  def latencyGauge[F[_]: Clock: Effect](config: MetricsReporter, ref: Ref[F, Option[(Long, Long)]]): Gauge[Long] = {
+    val resetDelta = config.period.toMillis * GaugeResetFactor
+    effectGauge(ref.get) {
+      case Some((acked, collectorTstamp)) =>
+        Clock[F].realTime(MILLISECONDS).map { now =>
+          if (now - acked < resetDelta)
+            acked - collectorTstamp
+          else
+            0L
```

Comment:
Do we really want to send a latency of 0 ? I would be in favor of not sending it instead (maybe by filtering out [here](https://github.com/snowplow/enrich/blob/streeter/dev/modules/fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/fs2/io/StatsDReporter.scala#L87-L92)).